### PR TITLE
Add an assert to ensure we not leak in the future.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should add an assert to ensure we not leak in the future as we not do any special handling if trySuccess(...) returns false.

Modifications:

Add assert.

Result:

More robust code.

### CODE REVIEW

File: resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java [hunk 1]
```diff
@@ -139,7 +140,13 @@ void resolve(final Promise<T> promise) {
                 public void operationComplete(Future<T> future) throws Exception {
                     Throwable cause = future.cause();
                     if (cause == null) {
-                        promise.trySuccess(future.getNow());
+                        T result = future.getNow();
+
+                        // Assert that we will not use any ReferenceCounted result here as trySuccess(...) may
+                        // return false in which case the ownership would not be transferred and special
+                        // handling would be required. This is not the case in our current implementation.
+                        assert !(result instanceof ReferenceCounted);
```
Comment (Scottmitch) [inline resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java]: `T` may also be a `List` if we are resolving all addresses. Not sure if we should account/care about collections if we are trying to add some protection for reference counting?

Comment (normanmaurer) [reply resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java]: I just wanted to keep it simple for now... I think we not need to also try to walk a List etc.

Comment (normanmaurer) [reply resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java]: @Scottmitch WDYT ?

**Comment by normanmaurer** [comment]
Let me just close this for now...

**Pull request closed without merging**